### PR TITLE
Templates bicerano

### PR DESCRIPTION
**Pull request by AdrianM0**
Added templates for bicerano + canonicalization of PSMILES:

Four failed test cases for the latter:

1 [\*]OC(=O)Oc1ccc(C(C)(C)c2ccc([\*])c(C)c2)cc1Cl  --- [\*]c1ccc(OC(=O)Oc2ccc(C([\*])(C)C)cc2Cl)c(C)c1
2 [\*]OC(=O)c1ccc(C(=O)Oc2ccc(C(C)(C)c3ccc([\*])c(C)c3)cc2C)cc1 --- [\*]c1ccc(OC(=O)c2ccc(C(=O)Oc3ccc(C([\*])(C)C)cc3C)cc2)c(C)c1
3 [\*]C(=O)c1ccc2c(c1)C(=O)N(c1cccc(C(=O)c3cccc(N4C(=O)c5ccc([\*])cc5C4=O)c3)c1)C2=O ---[\*]C(=O)c1cccc(N2C(=O)c3ccc(C(=O)c4ccc5c(c4)C(=O)N(c4cccc([\*])c4)C5=O)cc3C2=O)c1
4 [\*]c1ccc(Oc2ccc(-c3nc4ccc(Oc5ccc6nc([\*])c(-c7ccccc7)c(-c7ccccc7)c6c5)cc4c(-c4ccccc4)c3-c3ccccc3)cc2)cc1 [\*]Oc1ccc2nc(-
c3ccc(Oc4ccc(-c5nc6ccc([\*])cc6c(-c6ccccc6)c5-c5ccccc5)cc4)cc3)c(-c3ccccc3)c(-c3ccccc3)c2c1

### CODE REVIEW

File: data/tabular/bicerano_dataset/transform.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    clean_data["PSMILES"] = clean_data["PSMILES"].str.replace(
+        "[[*]]", "[*]", regex=False
+    )
+
+    clean_data["PSMILES"] = clean_data["PSMILES"].apply(
+        lambda smiles: canonicalize(smiles)
+    )
+
+    clean_data.to_csv("HT_MD_Polymers_clean.csv")
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/transform.py:43]:
```suggestion
    clean_data.to_csv("data_clean.csv")

```

**Comment by kjappelbaum** [comment]
do you know why there are merge conflicts? You didn't branch of the latest `main` when you started the PR?

**Comment by AdrianM0** [comment]
> do you know why there are merge conflicts? You didn't branch of the latest `main` when you started the PR?

not sure why, I did branch of the latest main

**Comment by AdrianM0** [comment]
might be because you pushed my changes without the intermediate commits that I made?

**Comment by kjappelbaum** [comment]
>might be because you pushed my previous changes without the intermediate commits that I made?

yeah, I merged at some point. Before that, I had fixed some linting issues. If you continued if your old branch without pulling, this will have caused the issue

File: data/tabular/bicerano_dataset/transform.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import pandas as pd
+from canonicalize_psmiles.canonicalize import canonicalize
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/transform.py:2]: can you add this dependency to the requirements?

File: data/tabular/bicerano_dataset/meta.yaml [hunk 3]
```diff
@@ -1,95 +1,122 @@
...
+    author = {Afzal, Mohammad Atif Faiz and Browning, Andrea R. and Goldberg, Alexander and Halls, Mathew D. and Gavartin, Jacob L. and Morisato,
+    Tsuguo and Hughes, Thomas F. and Giesen, David J. and Goose, Joseph E.},
+    title = {High-Throughput Molecular Dynamics Simulations and Validation of Thermophysical Properties of Polymers for Various Applications},
+    journal = {ACS Applied Polymer Materials},
+    volume = {3},
+    number = {2},
+    pages = {620-630},
+    year = {2021},
+    doi = {10.1021/acsapm.0c00524}}
+templates:
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:96]: A lot of nice prompts! 

I wonder if it makes sense to ask about standard deviations as the model will lack the context of how the simulations have been run. What do you think. 

In addition i'd perhaps replace "molecule" with "polymer" 

Did you also think of trying a prompt of the form "Question: What is a polymer with experimental glass transition temperature of {Tg_exp#} K and experimental density at 300 K of {rho_300K_exp#} g/cc. Answer: A polymer with {PSMILES__description} {PSMILES#}"

**Review by kjappelbaum — CHANGES_REQUESTED** [review]
Hey Adrian! Very nice additions. 

I'm not sure about asking about the standard deviation and would call polymer polymers. Perhaps I'd also add questions about designing polymers with desired properties.

**Comment by AdrianM0** [comment]
@kjappelbaum can we finalize this PR at some point?

**Comment by kjappelbaum** [comment]
Thanks for the reminder!

File: data/tabular/bicerano_dataset/meta.yaml
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:1]: I like your new templates!

**Review by kjappelbaum — APPROVED** [review]
nice addition, good to at least have some polymer data in ChemNLP. It is still very underrepresented, but this should already help a lot!

**Pull request merged**